File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_cpp_labs_and_solutions/decorator/solution/with_sugar.cpp
// with_sugar.cpp

#include <iostream>
#include "beverage.h"
#include "with_sugar.h"

with_sugar::with_sugar(beverage * a_successor)
: with(a_successor)
{
}

void with_sugar::prepare()
{
	with::prepare();
	std::cout << "Adding sugar.\n";
}

File: dp_labs_and_solutions/original/dp_labs_and_solutions/dp_java_labs_and_solutions/decorator/solution/WithSugar.java
// WithSugar.java

public class WithSugar extends With
{
	public WithSugar(Beverage aSuccessor)
	{
		super(aSuccessor);
	}

	public void prepare()
	{
		super.prepare();
		System.out.println("Adding sugar.");
	}
}
